Sources/LinuxPlatform/Linux.swift

```diff
@@ -41,48 +41,227 @@ public struct Linux: Platform {
+                "python3-lldb-13",
+                "tzdata",
+                "zlib1g-dev",
+            ]
+        case "amazonlinux2":
+            [
+                "binutils",
+                "gcc",
+                "git",
+                "unzip",
+                "glibc-static",
+                "gzip",
+                "libcurl-devel",
+                "libedit",
+                "libicu",
+                "libuuid",
+                "libxml2-devel",
+                "sqlite-devel",
+                "tar",
+                "tzdata",
+                "zlib-devel",
+            ]
+        case "ubi9":
+            [
+                "git",
+                "gcc-c++",
+                "libcurl-devel",
+                "libedit-devel",
+                "libuuid-devel",
+                "libxml2-devel",
+                "ncurses-devel",
+                "python3-devel",
+                "rsync",
+                "sqlite-devel",
+                "unzip",
+                "zip",
+            ]
+        default:
+            []
+        }
+
+        let manager: String? = switch platformName {
+        case "ubuntu1804":
+            "apt"
+        case "ubuntu2004":
+            "apt"
+        case "ubuntu2204":
+            "apt"
+        case "amazonlinux2":
+            "yum"
+        case "ubi9":
+            "yum"
+        default:
+            nil
+        }
+
+        if requireSignatureValidation {
+            guard (try? self.runProgram("gpg", "--version", quiet: true)) != nil else {
+                var msg = "gpg is not installed. "
+                if let manager = manager {
+                    msg += """
+                    You can install it by running this command as root:
+                        \(manager) -y install gpg
+                    """
+                } else {
+                    msg += "you can install gpg to get signature verifications of the toolchahins."
+                }
+                msg += "\n" + Self.skipVerificationMessage
+
+                throw Error(message: msg)
+            }
+
+            let foundKeys = (try? self.runProgram(
+                "gpg",
+                "--list-keys",
+                "swift-infrastructure@forums.swift.org",
+                "swift-infrastructure@swift.org",
+                quiet: true
+            )) != nil
+            if !foundKeys {
+                // Import the swift keys if they aren't here already
+                let tmpFile = self.getTempFilePath()
+                FileManager.default.createFile(atPath: tmpFile.path, contents: nil, attributes: [.posixPermissions: 0600])
+                defer {
+                    try? FileManager.default.removeItem(at: tmpFile)
+                }
+
+                guard let url = URL(string: "https://swift.org/keys/all-keys.asc") else {
+                    throw Error(message: "malformed URL to the swift gpg keys")
+                }
+
+                try await httpClient.downloadFile(url: url, to: tmpFile)
```

Comment:
I've found inside a vscode devcontainer `httpClient.downloadFile` fails to write to the tmpFile

Reply:
Worked out why octal literals are written `0o600`.